roxie/ccd/ccdprotocol.cpp

```diff
@@ -1977,6 +1977,11 @@ class RoxieSocketWorker : public ProtocolQueryWorker
                                     {
                                         fixedreq->addPropTree(iter->query().queryName(), LINK(&iter->query()));
                                     }
+                                    Owned<IAttributeIterator> aiter = queryPT->getAttributes();
```

Comment:
cut and paste: Should be reqIter->query().getttributes()

Reply:
Maybe - it's debatable whether attributes are common to every request in the array, or specified on each one. I can see arguments for both (or maybe even merge them).

You could also argue that for something like "statsToWorkunit" we should be add additional logic to make all the subqueries in the array write to the same workunit. But that would be something for a future enhancement.

Reply:
@afishbeck Do you have an opinion?

Reply:
I like the way it is now.  The attributes can be set at the array level, and/or overridden at the individual request.